lib/websocket/socket-event-emitter-adapter.ts

```diff
@@ -52,9 +52,13 @@ class SocketEventEmitterAdapter implements events.EventEmitter {
     }
 
     emit(event: string, ...args: any[]): boolean {
-        // Note that we want to stub calls to 'emit' because SocketIO.Socket.emit is specialized
-        // and does not follow the the standard NodeJS.EventEmitter interface.
-        assert(false, '“emit” should not be called on SocketEventEmitterAdapter');
-        return false;
+        if (this.emitter instanceof events.EventEmitter) {
+            return EVENTEMITTERPROTOTYPE.emit.apply(this.emitter, arguments);
+        } else {
+            // Note that we want to stub calls to 'emit' because SocketIO.Socket.emit is specialized
+            // and does not follow the the standard NodeJS.EventEmitter interface.
+            assert(false, '“emit” should not be called on SocketEventEmitterAdapter');
```

Comment:
Nit: change the message to indicate SocketIO since emit works for some types of SEEAs.